FrEee/Scripts/AI/Vanguard/AI.csx

```diff
@@ -0,0 +1,122 @@
+    MinistryOfColonization MinistryOfColonization;
+    MinistryOfInfrastructure MinistryOfInfrastructure; 
+
+    public Runner()
+    {
+        BasicResearch = new BasicResearch();
+        MinistryOfShipDesign = new MinistryOfShipDesign();
+        MinistryOfConstruction = new MinistryOfConstruction();
+        MinistryOfColonization = new MinistryOfColonization();
+        MinistryOfInfrastructure = new MinistryOfInfrastructure(); 
+    }
+
+    public Empire Run(Empire Domain, Galaxy Context)
+    {
+        try
+        {
+            galaxy = Context;
+            empire = Domain;
+            PlanManager.Empire = empire;
+            PlanManager.Unpack(); 
+
+            if (empire.EnabledMinisters.ContainsKey("Facility Management"))
+            {
+                //split up the facility management into construction, upgrades, and replacement? make it its own category? 
+                var colonyManagement = empire.EnabledMinisters["Facility Management"];
+                if (colonyManagement.Contains("Facility Construction"))
+                    MinistryOfInfrastructure.Run(empire, galaxy);
+            }
+
+            if (empire.EnabledMinisters.ContainsKey("Design Management"))
+            {
+                MinistryOfShipDesign.Run(empire, galaxy); 
+            }
+
+            if (empire.EnabledMinisters.ContainsKey("Empire Management"))
+            {
+                var managementMinisters = empire.EnabledMinisters["Empire Management"]; 
+                if (managementMinisters.Contains("Research"))
+                    BasicResearch.Run(empire, galaxy);
+            }
+
+
+            if (empire.EnabledMinisters.ContainsKey("Vehicle Management"))
+                if (empire.EnabledMinisters["Vehicle Management"].Contains("Ship Construction"))
+                    MinistryOfConstruction.HandleNewlyConstructedShips(empire, galaxy);
+
+            if (empire.EnabledMinisters.ContainsKey("Expansion"))
+            {
+                var managementMinisters = empire.EnabledMinisters["Expansion"];
+
+                if (managementMinisters.Contains("Colonization"))
+                    MinistryOfColonization.Run(empire, galaxy);
+            }
+
+            if (empire.EnabledMinisters.ContainsKey("Vehicle Management"))
+            {
+                var managementMinisters = empire.EnabledMinisters["Vehicle Management"];
+               
+                if (managementMinisters.Contains("Ship Construction"))
+                    MinistryOfConstruction.ConstructShips(empire, galaxy);
+                if (managementMinisters.Contains("Unit Construction"))
+                    MinistryOfConstruction.ConstructUnits(empire, galaxy); 
+            }
+
+            PlanManager.RemoveCompleted(); 
+            PlanManager.Pack(); 
+        }
+        catch (Exception e)
+        {
+            throw; 
+        }
+        return empire;  
+        
+    }
+
+
+    //void SetResearch()
```

Comment:
If you're not using this method, you might as well delete it.

Reply:
Especially as I am using it, but in another class entirely.